### PR TITLE
Add necessary AppConfigs and handle the reconfiguration during ranging

### PR DESCRIPTION
**Pull request by bobwzy**
- Add necessary app configs. Handle reconfiguration during ranging.
It is allowed to set app config in the case that the device is in ranging and
`reconfigure` is called from the host.

### CODE REVIEW

**Comment by SilverBzH** [comment]
Please can you rebase your PR on top of main ?

File: src/session.rs [hunk 1]
```diff
@@ -96,14 +96,152 @@ enum UpdateMulticastListAction {
...
+enum RangeDataNtfConfig {
+    Disable = 0x00,
+    Enable = 0x01,
+    EnableProximityLevelTrig = 0x02,
+    EnableAoaLevelTrig = 0x03,
+    EnableProximityAoaLevelTrig = 0x04,
+    EnableProximityEdgeTrig = 0x05,
+    EnableAoaEdgeTrig = 0x06,
+    EnableProximityAoaEdgeTrig = 0x07,
+}
```
Comment (SilverBzH) [inline src/session.rs:236]: I think this should be in pdl directly, is it coming from to the spec ?

Comment (bobwzy) [reply src/session.rs:236]: It is not in pdl. They are from UCI spec and are possible values for each app config.

File: src/session.rs [hunk 2]
```diff
@@ -323,13 +631,18 @@ impl Session {
         assert_eq!(self.id, cmd.get_session_token());
         assert_eq!(self.session_type, SessionType::FiraRangingSession);
 
-        let (status, invalid_parameters) = if self.state != SessionState::SessionStateInit {
+        let (status, invalid_parameters) = if self.state != SessionState::SessionStateInit
```
Comment (hchataing) [inline src/session.rs:634]: > The Host can set some of the APP Configuration Parameters while in the SESSION_STATE_ACTIVE
> state as well. The UWBS shall respond with a SESSION_SET_APP_CONFIG_RSP setting Status to
> STATUS_ERROR_SESSION_ACTIVE if the Host tries to apply configurations which are not allowed to
> set during the SESSION_STATE_ACTIVE Session State. Host shall use RANGE_STOP_CMD to stop
> the active UWB session before applying the new set of APP Configuration Parameters which are not
> permitted to change during SESSION_STATE_ACTIVE.

1. Can you have a separate check for config entries that do against this requirement ?
2. I don't see it stated that the ranging session shall be stopped when invalid parameters are passed; only that an error response is returned, and that the host should stop ranging before retrying

Comment (bobwzy) [reply src/session.rs:638]: Done. According to the spec, only `AoaResultReq` cannot be changed when the status is `Active`.

File: src/session.rs [hunk 3]
```diff
@@ -323,13 +631,18 @@ impl Session {
...
+            && self.state != SessionState::SessionStateActive
+        {
             (StatusCode::UciStatusRejected, Vec::new())
         } else {
             let mut app_config = self.app_config.clone();
             let invalid_parameters = app_config.extend(cmd.get_tlvs());
             if invalid_parameters.is_empty() {
                 self.app_config = app_config;
+                if self.state == SessionState::SessionStateActive {
```
[the remark this answers is not in the document]
Comment (bobwzy) [reply src/session.rs]: It was a mistake. So I removed this change in the latest verison.

[the remark this answers is not in the document]
Comment (hchataing) [reply src/session.rs]: I missed the modification, thanks

**Review by hchataing — APPROVED** [review]

**Pull request merged**